src/ctap/mod.rs

```diff
@@ -228,11 +235,15 @@ where
         credential_id: Vec<u8>,
         rp_id_hash: &[u8],
     ) -> Result<Option<PublicKeyCredentialSource>, Ctap2StatusCode> {
-        if credential_id.len() != ENCRYPTED_CREDENTIAL_ID_SIZE {
+        let has_cred_random = if credential_id.len() == CREDENTIAL_ID_BASE_SIZE {
```

Comment:
nit: It looks odd to me to see 3 branches and only 2 resulting in an assigned value while the other one will early stop the function.

Reply:
Good point, should be more readable with a `match`.